src/commons/scheduler.py

```diff
@@ -84,13 +92,15 @@ async def schedule_sessions(test_plan: str, test_plan_value: dict, common_params
                                         start_time=each["start_time"].total_seconds(), **params))
         else:
             raise NotImplementedError(f"Tool is not supported: {each['tool']}")
-    done, pending = await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
-    LOGGER.critical("Terminating process & pending task: %s", process_name)
-    for task in pending:
-        task.cancel()
-    LOGGER.critical(done)
-    for task in done:
-        task.result()
+    done, pending = await asyncio.wait(tasks, return_when=asyncio.FIRST_EXCEPTION)
+    if pending:
+        LOGGER.critical("Terminating process: %s & pending tasks: %s", process_name, pending)
+        for task in pending:
+            task.cancel()
+    if done:
+        LOGGER.info("Completed tasks: %s", done)
+        for task in done:
+            task.result()
```

Comment:
Please remove this for loop. It doesn't have any effect.

Reply:
This is old code so I'll keep it as is.